src/main/java/seedu/address/logic/commands/DuplicateTransactionCommand.java

```diff
@@ -1,54 +1,82 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIMESTAMP;
 
 import java.util.List;
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
+import seedu.address.logic.commands.EditTransactionCommand.EditTransactionDescriptor;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.transaction.Transaction;
 
 /**
- * Duplicates a transaction identified using it's displayed index from the address book.
+ * Duplicates a transaction identified using its displayed index from the address book.
+ * Optional edits can be made to the duplicated transaction.
+ * Timestamp will be set to the current time if not specified.
  */
 public class DuplicateTransactionCommand extends Command {
 
     public static final String COMMAND_WORD = "duplicateTransaction";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-        + ": Duplicates the transaction identified by the index number used in the displayed transaction list.\n"
-        + "Parameters: INDEX (must be a positive integer)\n"
-        + "Example: " + COMMAND_WORD + " 1";
-
-    public static final String MESSAGE_DUPLICATE_TRANSACTION_SUCCESS = "Duplicated Transaction: %1$s";
+        + ": Duplicates the transaction identified by the index number with optional edits. "
+        + "Timestamp will be set to the current time if not specified.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + "[" + PREFIX_COST + "COST] "
+        + "[" + PREFIX_DESCRIPTION + "DETAILS] "
+        + "[" + PREFIX_NAME + "PAYEE NAME] "
+        + "[" + PREFIX_TIMESTAMP + "TIMESTAMP]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_COST + "10.00 "
+        + PREFIX_DESCRIPTION + "Bought a book "
+        + PREFIX_NAME + "John Doe "
+        + PREFIX_TIMESTAMP + "2020-01-01 12:00";
+
+    public static final String MESSAGE_DUPLICATE_TRANSACTION_SUCCESS = "New duplicated transaction added: %1$s";
+
+    public static final String MESSAGE_TRANSACTION_NOT_RELEVANT =
+        "The duplicated transaction does not affect your balances";
 
     private final Index targetIndex;
+    private final EditTransactionDescriptor duplicateTransactionDescriptor;
```

Comment:
Now that `editTransactionDescriptor` is also used by another class, should we create a superclass for editing/duplicating transactions? Because `editTransactionDescriptor` is an inner class residing in `editTransactionCommand`, it doesn't seem so good that another class would have access to it.

Reply:
It is one possible way to do so, but I'm not sure what would be an appropriate class for both to inherit from.

Another way I can think of is to isolate the `TransactionDescriptor` as a util class on its own.